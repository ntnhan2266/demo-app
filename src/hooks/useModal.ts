import { useState, useCallback } from 'react';

interface IModalHook {
  isOpen: boolean;
  openModal: () => void;
  closeModal: () => void;
}

const useModal = (): IModalHook => {
  const [isOpen, setIsOpen] = useState(false);

  const openModal = useCallback(() => {
    setIsOpen(true);
  }, []);

  const closeModal = useCallback(() => {
    setIsOpen(false);
  }, []);

  return { isOpen, openModal, closeModal };
};

export default useModal;
