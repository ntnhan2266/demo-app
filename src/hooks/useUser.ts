import { useContext } from "react";
import { UserContext } from "@/contexts/UserContext";

// Custom hook to use the user context
export const useUser = () => {
  const context = useContext(UserContext);

  if (!context) {
    throw new Error('useUser must be used within a UserProvider');
  }

  return context;
};
