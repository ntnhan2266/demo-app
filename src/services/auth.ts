import { IUser } from '@/interfaces/user';
import { saveAuthUserToLocalStorage } from '@/utils/auth-storage';
import {
  loginUser as mockLoginUser,
  registerUser as mockRegisterUser,
  updateUser as mockUpdateUser,
  updatePassword as mockUpdatePassword,
} from '@/utils/mock-user-database';

export const registerUser = (user: IUser): boolean => {
  // Just call mock function to simulate api call
  return mockRegisterUser(user);
};

export const loginUser = (emailOrPhone: string, password: string): IUser | null => {
  const matchedUser = mockLoginUser(emailOrPhone, password);

  if (matchedUser) {
    // Save auth user info
    saveAuthUserToLocalStorage(matchedUser);

    // Return user info upon successful login
    return matchedUser;
  }

  return null; // Return null for unsuccessful login
};

export const updateUserInfo = (emailOrPhone: string, updatedInfo: Partial<IUser>): IUser | null => {
  // Just call mock function to simulate api call
  const updatedUser = mockUpdateUser(emailOrPhone, updatedInfo);

  if (updatedUser) {
    // Update auth user info
    saveAuthUserToLocalStorage(updatedUser);

    // Return user info upon successful login
    return updatedUser;
  }

  return null;
};

export const updateUserPassword = (emailOrPhone: string, oldPassword: string, newPassword: string): boolean => {
  // Just call mock function to simulate api call
  return mockUpdatePassword(emailOrPhone, oldPassword, newPassword);
};
