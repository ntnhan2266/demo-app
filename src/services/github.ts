import { IGetGithubReposParams, IGithubRepo } from '@/interfaces/api/github';
import createAxiosInstance from '@/services/http-client';
import { snakeCaseKeys } from '@/utils/convert';

const API_PATH = 'https://api.github.com';

const apiClient = createAxiosInstance({
  baseURL: API_PATH,
  headers: {
    Accept: 'application/vnd.github+json',
  },
});

export const getListReposByUsername = async (params: IGetGithubReposParams) => {
  try {
    // Destructure the params
    const { username, page = 1, perPage = 20, sort = 'updated_at', direction = 'desc' } = params;

    // Make the request to the GitHub API
    const response = await apiClient.get(`/users/${username}/repos`, {
      // Adjusting the parameter name based on the interface
      params: snakeCaseKeys({
        page,
        perPage,
        sort,
        direction,
      }),
    });

    // Convert to Array<IGithubRepo>
    const repos: Array<IGithubRepo> = response.data as Array<IGithubRepo>;
    return repos;
  } catch (error) {
    // Handle errors here, e.g., log the error, show a user-friendly message, etc.
    console.error('Error fetching GitHub repositories:', error);
    // Rethrow the error for the calling code to handle
    throw error;
  }
};
