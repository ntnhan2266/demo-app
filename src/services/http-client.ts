import { camelCaseKeys } from '@/utils/convert';
import axios, { AxiosInstance } from 'axios';

// Timeout 1 minute
const REQUEST_TIMEOUT = 60 * 1000;

interface IAxiosInstanceOptions {
  baseURL: string;
  headers?: Record<string, string>;
}

/**
 * Create and configure an Axios instance with interceptors for handling requests and responses.
 * @param {AxiosInstanceOptions} options - Configuration options for the Axios instance.
 * @returns {AxiosInstance} - Configured Axios instance.
 */
const createAxiosInstance = ({ baseURL = '', headers }: IAxiosInstanceOptions): AxiosInstance => {
  const instance = axios.create({
    baseURL,
    timeout: REQUEST_TIMEOUT,
    headers: {
      'Content-Type': 'application/json',
      ...headers,
    },
  });

  instance.interceptors.request.use(
    (config) => {
      // Modify the request config here, such as adding authentication headers
      // config.headers.Authorization = `Bearer ${getToken()}`;
      return config;
    },
    (error) => {
      return Promise.reject(error);
    },
  );

  instance.interceptors.response.use(
    (response) => {
      // Modify the response here before it reaches the calling component
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      const modifiedResponseData = camelCaseKeys(response.data as Record<string, any>);
      return { ...response, data: modifiedResponseData };
    },
    (error) => {
      // You can handle errors globally here, such as redirecting to an error page
      return Promise.reject(error);
    },
  );

  return instance;
};

export default createAxiosInstance;
