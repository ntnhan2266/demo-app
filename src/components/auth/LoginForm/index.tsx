import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import { useNavigate } from 'react-router-dom';
import { loginUser } from '@/services/auth';
import Input from '@/components/common/Input';
import Button from '@/components/common/Button';
import { ROUTE_PATH } from '@/constants/route-path';
import { isValidContact } from '@/utils/validation';
import { useUser } from '@/hooks/useUser';

interface ILoginFormInputs {
  emailOrPhone: string;
  password: string;
}

const LoginForm: React.FC = (): React.ReactElement => {
  const navigate = useNavigate();
  const {
    register,
    handleSubmit,
    reset,
    formState: { errors },
  } = useForm<ILoginFormInputs>();
  const { setUser } = useUser();
  const [successMessage, setSuccessMessage] = useState<string | null>(null);
  const [errorMessage, setErrorMessage] = useState<string | null>(null);

  const onSubmit = (values: ILoginFormInputs): void => {
    const user = loginUser(values.emailOrPhone, values.password);

    if (!user) {
      setErrorMessage('Login failed. Invalid credentials.');
      return;
    }

    // For simplicity, just display a success message
    setSuccessMessage('Login successful!');

    // Update authenticated user data into context
    setUser(user);

    // Reset the form
    // Note: Use reset() if you need to clear the form fields
    reset();

    // Navigate to the dashboard or any other page after login
    navigate(ROUTE_PATH.DASHBOARD_UPDATE_INFO);
  };

  return (
    <div data-testid='login-form' className='max-w-md mx-auto mt-8 p-6 bg-white rounded-md shadow'>
      <h2 className='text-2xl font-bold mb-4 uppercase'>Login to your account</h2>
      <form onSubmit={handleSubmit(onSubmit)}>
        <Input
          {...register('emailOrPhone', {
            required: 'Email or phone is required.',
            validate: (val: string) => isValidContact(val) || 'Invalid email or phone.',
          })}
          errorMessage={errors.emailOrPhone?.message}
          type='text'
          label='Email or phone'
          placeholder='Enter your email or phone number'
          wrapperClass='mb-4'
          data-testid='email-or-phone-input'
          inputClasses='w-full md:w-full'
          id='email'
          required
        />
        <Input
          {...register('password', {
            required: 'Password is required.',
          })}
          errorMessage={errors.password?.message}
          type='password'
          label='Password'
          placeholder='Enter your password'
          wrapperClass='mb-4'
          data-testid='password-input'
          inputClasses='w-full md:w-full'
          id='password'
          required
        />
        <Button data-testid='submit-button' label='Login' type='submit' fullWidth />
      </form>
      {successMessage && (
        <div data-testid className='mt-4 p-3 bg-green-200 text-green-800 rounded-md'>
          {successMessage}
        </div>
      )}
      {errorMessage && (
        <div data-testid='error-message' className='mt-4 p-3 bg-red-200 text-red-800 rounded-md'>
          {errorMessage}
        </div>
      )}
    </div>
  );
};

export default LoginForm;
