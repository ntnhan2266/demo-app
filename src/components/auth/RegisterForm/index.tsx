import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import Button from '@/components/common/Button';
import Input from '@/components/common/Input';
import { isValidContact } from '@/utils/validation';
import { registerUser } from '@/services/auth';
import ErrorModal from '@/components/common/ErrorModal';
import SuccessModal from '@/components/common/SuccessModal';

interface IFormInputs {
  firstName: string;
  lastName: string;
  emailOrPhone: string;
  password: string;
  confirmPassword: string;
}

const RegisterForm: React.FC = (): React.ReactElement => {
  const {
    register,
    handleSubmit,
    reset,
    watch,
    setError,
    formState: { errors },
  } = useForm<IFormInputs>({
    defaultValues: {
      firstName: '',
      lastName: '',
      emailOrPhone: '',
      password: '',
      confirmPassword: '',
    },
  });
  const [isErrorModalOpen, setIsErrorModalOpen] = useState(false);
  const [isSuccessModalOpen, setIsSuccessModalOpen] = useState(false);

  const closeErrorModal = (): void => {
    // Close the error modal
    setIsErrorModalOpen(false);
  };

  const closeSuccessModal = (): void => {
    // Close the success modal
    setIsSuccessModalOpen(false);
  };

  const onSubmit = (values: IFormInputs): void => {
    const result = registerUser(values);
    if (!result) {
      // Registration failed
      setIsErrorModalOpen(true);
      setError('emailOrPhone', {
        message: 'This email or phone is already in use.',
      });
      return;
    }

    // Registration successful
    setIsSuccessModalOpen(true);

    // Reset the form
    reset();
  };

  return (
    <div data-testid='register-form' className='max-w-md mx-auto mt-8 p-6 bg-white rounded-md shadow'>
      <h2 className='text-2xl font-bold mb-4 uppercase'>Create your account</h2>
      <form onSubmit={handleSubmit(onSubmit)}>
        <Input
          {...register('firstName', { required: 'First name is required.' })}
          errorMessage={errors.firstName?.message}
          type='text'
          label='First name'
          placeholder='Enter your first name'
          wrapperClass='mb-4'
          data-testid='first-name-input'
          inputClasses='w-full md:w-full'
          required
        />
        <Input
          {...register('lastName', { required: 'Last name is required.' })}
          errorMessage={errors.lastName?.message}
          type='text'
          label='Last name'
          placeholder='Enter your last name'
          wrapperClass='mb-4'
          data-testid='last-name-input'
          inputClasses='w-full md:w-full'
          required
        />
        <Input
          {...register('emailOrPhone', {
            required: 'Email or phone is required.',
            validate: (val: string) => isValidContact(val) || 'Invalid email or phone.',
          })}
          errorMessage={errors.emailOrPhone?.message}
          type='text'
          name='emailOrPhone'
          label='Email or phone'
          placeholder='Enter either your email or phone number'
          wrapperClass='mb-4'
          data-testid='email-or-phone-input'
          inputClasses='w-full md:w-full'
          required
        />
        <Input
          {...register('password', {
            required: 'Password is required.',
            pattern: {
              value: /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[!@#$%^&*(),.?":{}|<>])[A-Za-z\d!@#$%^&*(),.?":{}|<>]{12,}$/,
              message:
                'Password must be strong (include uppercase, lowercase, number, special character, and at least 12 characters).',
            },
          })}
          errorMessage={errors.password?.message}
          type='password'
          label='Password'
          placeholder='Enter your password'
          wrapperClass='mb-4'
          data-testid='password-input'
          inputClasses='w-full md:w-full'
          required
        />
        <Input
          {...register('confirmPassword', {
            required: 'Confirm password is required.',
            validate: (val: string) => {
              if (watch('password') != val) {
                return 'Your passwords do not match.';
              }
            },
          })}
          errorMessage={errors.confirmPassword?.message}
          type='password'
          label='Confirm password'
          placeholder='Confirm your password'
          wrapperClass='mb-4'
          data-testid='confirm-password-input'
          inputClasses='w-full md:w-full'
          required
        />
        <Button data-testid='submit-button' label='Register' type='submit' fullWidth />
      </form>
      <ErrorModal
        title='Failure!'
        content='There was an error during the registration process. Please try again.'
        isOpen={isErrorModalOpen}
        onClose={closeErrorModal}
      />
      <SuccessModal
        title='Success!'
        content='Your account has been successfully registered.'
        isOpen={isSuccessModalOpen}
        onClose={closeSuccessModal}
      />
    </div>
  );
};

export default RegisterForm;
