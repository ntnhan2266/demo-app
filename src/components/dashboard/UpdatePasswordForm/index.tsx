import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import Button from '@/components/common/Button';
import Input from '@/components/common/Input';
import ErrorModal from '@/components/common/ErrorModal';
import SuccessModal from '@/components/common/SuccessModal';
import { updateUserPassword } from '@/services/auth';
import { useUser } from '@/hooks/useUser';

interface IFormInputs {
  currentPassword: string;
  newPassword: string;
  confirmNewPassword: string;
}

const UpdateUserPasswordForm: React.FC = (): React.ReactElement => {
  const { user } = useUser();
  const {
    register,
    handleSubmit,
    reset,
    watch,
    formState: { errors },
  } = useForm<IFormInputs>({
    defaultValues: {
      currentPassword: '',
      newPassword: '',
      confirmNewPassword: '',
    },
  });
  const [errorMessage, setErrorMessage] = useState<string | null>(null);
  const [isErrorModalOpen, setIsErrorModalOpen] = useState(false);
  const [isSuccessModalOpen, setIsSuccessModalOpen] = useState(false);

  const closeErrorModal = (): void => {
    setIsErrorModalOpen(false);
  };

  const closeSuccessModal = (): void => {
    setIsSuccessModalOpen(false);
  };

  const onSubmit = (values: IFormInputs): void => {
    setErrorMessage(null);

    if (!user) {
      setErrorMessage('User not found');
      setIsErrorModalOpen(true);
      return;
    }

    const result = updateUserPassword(user.emailOrPhone, values.currentPassword, values.newPassword);

    if (!result) {
      // Password update failed
      setIsErrorModalOpen(true);
      setErrorMessage('Incorrect current password.');
      return;
    }

    // Password update successful
    setIsSuccessModalOpen(true);

    // Reset the form
    reset();
  };

  return (
    <div data-testid='update-user-password-form'>
      <h2 className='text-2xl font-bold mb-8 uppercase'>Update your password</h2>
      <form onSubmit={handleSubmit(onSubmit)}>
        <Input
          {...register('currentPassword', { required: 'Current password is required.' })}
          errorMessage={errors.currentPassword?.message}
          type='password'
          label='Current Password'
          placeholder='Enter your current password'
          wrapperClass='mb-4'
          data-testid='current-password-input'
          required
        />
        <Input
          {...register('newPassword', {
            required: 'New password is required',
            pattern: {
              value: /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[!@#$%^&*(),.?":{}|<>])[A-Za-z\d!@#$%^&*(),.?":{}|<>]{12,}$/,
              message:
                'Password must be strong (include uppercase, lowercase, number, special character, and at least 12 characters).',
            },
          })}
          errorMessage={errors.newPassword?.message}
          type='password'
          label='New Password'
          placeholder='Enter your new password'
          wrapperClass='mb-4'
          data-testid='new-password-input'
          required
        />
        <Input
          {...register('confirmNewPassword', {
            required: 'Confirm new password is required.',
            validate: (val: string) => {
              if (watch('newPassword') !== val) {
                return 'Your new passwords do not match.';
              }
            },
          })}
          errorMessage={errors.confirmNewPassword?.message}
          type='password'
          label='Confirm New Password'
          placeholder='Confirm your new password'
          wrapperClass='mb-4'
          data-testid='confirm-password-input'
          required
        />
        <Button data-testid='submit-button' label='Update Password' type='submit' className='w-full lg:w-96' />
      </form>
      <ErrorModal
        title='Failure!'
        content={errorMessage || 'There was an error during the password update process. Please try again.'}
        isOpen={isErrorModalOpen}
        onClose={closeErrorModal}
      />
      <SuccessModal
        title='Success!'
        content='Your password has been successfully updated.'
        isOpen={isSuccessModalOpen}
        onClose={closeSuccessModal}
      />
    </div>
  );
};

export default UpdateUserPasswordForm;
