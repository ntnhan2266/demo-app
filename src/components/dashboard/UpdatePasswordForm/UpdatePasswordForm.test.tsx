import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import UpdatePasswordForm from '@/components/dashboard/UpdatePasswordForm';
import { useUser } from '@/hooks/useUser';
import { updateUserPassword } from '@/services/auth';
import renderer from 'react-test-renderer';

jest.mock('@/hooks/useUser');
jest.mock('@/services/auth');

describe('UpdatePasswordForm component', () => {
  const mockUseUser = useUser as jest.Mock;
  const mockUpdateUserPassword = updateUserPassword as jest.Mock;

  beforeEach(() => {
    mockUseUser.mockReturnValue({ user: { emailOrPhone: 'test@example.com' } });
    render(<UpdatePasswordForm />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should match the snapshot', () => {
    const tree = renderer.create(<UpdatePasswordForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the form with the correct inputs', () => {
    const currentPasswordInput = screen.getByTestId('current-password-input');
    const newPasswordInput = screen.getByTestId('new-password-input');
    const confirmNewPasswordInput = screen.getByTestId('confirm-password-input');
    const updateButton = screen.getByTestId('submit-button');

    expect(currentPasswordInput).toBeInTheDocument();
    expect(newPasswordInput).toBeInTheDocument();
    expect(confirmNewPasswordInput).toBeInTheDocument();
    expect(updateButton).toBeInTheDocument();
  });

  test('displays error messages when form is submitted with invalid data', async () => {
    const updateButton = screen.getByTestId('submit-button');
    fireEvent.click(updateButton);

    // Ensure that error messages are displayed for required fields
    expect(await screen.findByText('Current password is required.')).toBeInTheDocument();
    expect(await screen.findByText('New password is required')).toBeInTheDocument();
    expect(await screen.findByText('Confirm new password is required.')).toBeInTheDocument();
  });

  test('displays error message when current password is incorrect', async () => {
    mockUpdateUserPassword.mockReturnValue(false);

    const currentPasswordInput = screen.getByTestId('current-password-input');
    const newPasswordInput = screen.getByTestId('new-password-input');
    const confirmNewPasswordInput = screen.getByTestId('confirm-password-input');
    fireEvent.change(currentPasswordInput, {
      target: {
        value: 'qwerQWER1234!@',
      },
    });
    fireEvent.change(newPasswordInput, {
      target: {
        value: 'qwerQWER1234!@@',
      },
    });
    fireEvent.change(confirmNewPasswordInput, {
      target: {
        value: 'qwerQWER1234!@@',
      },
    });

    const updateButton = screen.getByTestId('submit-button');
    fireEvent.click(updateButton);

    // Ensure that an error message for incorrect current password is displayed
    expect(await screen.findByText('Incorrect current password.')).toBeInTheDocument();
  });

  test('calls updateUserPassword function when form is submitted with valid data', async () => {
    mockUpdateUserPassword.mockReturnValue(true);

    // Fill in the form
    const currentPasswordInput = screen.getByTestId('current-password-input');
    const newPasswordInput = screen.getByTestId('new-password-input');
    const confirmNewPasswordInput = screen.getByTestId('confirm-password-input');
    const updateButton = screen.getByTestId('submit-button');

    fireEvent.change(currentPasswordInput, {
      target: {
        value: 'qwerQWER1234!@',
      },
    });
    fireEvent.change(newPasswordInput, {
      target: {
        value: 'qwerQWER1234!@@',
      },
    });
    fireEvent.change(confirmNewPasswordInput, {
      target: {
        value: 'qwerQWER1234!@@',
      },
    });

    fireEvent.click(updateButton);

    // Wait for the success modal to appear
    await waitFor(() => {
      expect(mockUpdateUserPassword).toHaveBeenCalledWith('test@example.com', 'qwerQWER1234!@', 'qwerQWER1234!@@');
      expect(screen.getByText('Your password has been successfully updated.')).toBeInTheDocument();
    });
  });
});
