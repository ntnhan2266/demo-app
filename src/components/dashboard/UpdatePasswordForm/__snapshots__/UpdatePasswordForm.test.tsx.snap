// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdatePasswordForm component should match the snapshot 1`] = `
<div
  data-testid="update-user-password-form"
>
  <h2
    className="text-2xl font-bold mb-4"
  >
    Update your password
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div
      className="mb-4"
    >
      <label
        className="block text-sm font-semibold mb-1"
        htmlFor="Current Password"
      >
        Current Password
        <sup
          className="text-red-600"
        >
          *
        </sup>
      </label>
      <input
        className="border rounded-md px-3 py-2 focus:outline-none transition duration-300 w-96 max-w-full  border-gray-400"
        data-testid="current-password-input"
        disabled={false}
        name="currentPassword"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter your current password"
        type="password"
      />
    </div>
    <div
      className="mb-4"
    >
      <label
        className="block text-sm font-semibold mb-1"
        htmlFor="New Password"
      >
        New Password
        <sup
          className="text-red-600"
        >
          *
        </sup>
      </label>
      <input
        className="border rounded-md px-3 py-2 focus:outline-none transition duration-300 w-96 max-w-full  border-gray-400"
        data-testid="new-password-input"
        disabled={false}
        name="newPassword"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter your new password"
        type="password"
      />
    </div>
    <div
      className="mb-4"
    >
      <label
        className="block text-sm font-semibold mb-1"
        htmlFor="Confirm New Password"
      >
        Confirm New Password
        <sup
          className="text-red-600"
        >
          *
        </sup>
      </label>
      <input
        className="border rounded-md px-3 py-2 focus:outline-none transition duration-300 w-96 max-w-full  border-gray-400"
        data-testid="confirm-password-input"
        disabled={false}
        name="confirmNewPassword"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Confirm your new password"
        type="password"
      />
    </div>
    <button
      className="px-4 py-2 rounded-md focus:outline-none w-96 bg-blue-500 text-white"
      data-testid="submit-button"
      disabled={false}
      type="submit"
    >
      Update Password
    </button>
  </form>
</div>
`;
